datasource/src/DataSource.ts

```diff
@@ -107,8 +110,17 @@ export class DataSource extends DataSourceApi<SignalKQuery, SignalKDataSourceOpt
     const result = new Observable<DataQueryResponse>((subscriber) => {
       let lastStreamingValueTimestamp = 0;
 
-      const dataframe = new DualDataFrame(options.targets.map(
-        target => `${target.path}:${target.aggregate}`), 1000);
+      // const dataframe = new DualDataFrame(options.targets.map(
+      //   target => `${target.path}:${target.aggregate}`), 1000);
+      const dataframe = new CircularDataFrame({
```

Comment:
Have you considered using [StreamingDataFrame](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/dataframe/StreamingDataFrame.ts) with the related [DataFrameJSON](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/dataframe/DataFrameJSON.ts) instead of a Circular buffer? I'm wondering why grafana-data provides this class while it's extending MutableDataFrame that's deprecated. Also it's using CircularVector within which is also deprecated - simple arrays should be used instead. I wanted to bring this up since this one might soon be deprecated too.

Reply:
StreamingDataFrame is marked as alpha, so kinda stopped looking. Don't want to study & adopt something only to discover they went some other route.